src/client/proxy/http.h

```diff
@@ -210,73 +156,183 @@ class HTTPProxyHandler
+/// @class HTTPProtocol
+/// @brief defines protocol; and read from socket algorithm
+class HTTPProtocol : public std::enable_shared_from_this<HTTPProtocol>{
+ public:
+  std::string m_RequestLine, m_HeaderLine, m_Request, m_Body,
+    m_URL, m_Method, m_Version,  m_Path;
+  std::vector<std::string> m_Headers;
+  std::string m_Host, m_UserAgent;
+  std::string m_Address;
 
-  /// @brief Handles stream created by service through proxy handler
-  void HandleStreamRequestComplete(
-      std::shared_ptr<kovri::client::Stream> stream);
+  boost::asio::streambuf m_Buffer;
+  boost::asio::streambuf m_BodyBuffer;
+  std::vector<std::pair<std::string, std::string>> m_HeaderMap;
+  /// @brief Data for incoming request
+  std::uint16_t m_Port;
 
-  /// @enum State
-  /// @brief Parsing state
-  /// @note Only GET method currently supported
-  enum State : const std::uint8_t {
-    /// @var get_method
-    /// @brief Method sent in request
-    get_method,
-    /// @var get_url
-    /// @brief URL sent in request
-    get_url,
-    /// @var get_http_version
-    /// @brief HTTP version
-    get_http_version,
-    /// @var host
-    /// @brief Host: sent in request
-    host,
-    /// @var useragent
-    /// @brief User-Agent: sent in request
-    useragent,
-    /// @var newline
-    /// @brief Newline in request
-    newline,
-    /// @var done
-    /// @brief Done with request
-    done,
-  } m_State;
-
-  /// @brief Sets state set by handled data
-  void SetState(
-      const State& state);
+  /// @var m_JumpService
+  /// @brief Address helpers for base64 jump service
+  const std::array<std::string, 4> m_JumpService {
+  {
+    "?i2paddresshelper=",
+    "&i2paddresshelper=",
+    "?kovrijumpservice=",
+    "&kovrijumpservice=",
+  }
+  };
+  HTTPResponse m_ErrorResponse;
+  HTTPProtocol():m_Port(0), m_ErrorResponse(HTTPResponseCodes::status_t::ok) {
+  }
+  enum msg_t {
+    response,
+    request
+  };
+  /// @brief loads variables in class;
+  /// @param buf
+  /// @param len
+  /// return bool
+  bool HandleData(const std::string  &buf);
 
-  /// @brief Processes original request: extracts, validates,
-  ///   calls jump service, appends original request
-  /// @return true on success
-  bool CreateHTTPRequest(
-      uint8_t *buf,
-      std::size_t len);
+  /// @brief Parses path for base64 address, inserts into address book
+  /// need to change to boolean return for testing
+  void HandleJumpService();
 
   /// @brief Performs regex, sets address/port/path, validates version
   ///   on request sent from user
   /// @return true on success
   bool ExtractIncomingRequest();
 
-  /// @brief Parses path for base64 address, inserts into address book
-  void HandleJumpService();
+  /// @brief Processes original request: extracts, validates,
+  ///   calls jump service, appends original request
+  /// @return true on success
+  bool CreateHTTPRequest();
+
+  const unsigned int HEADERBODY_LEN = 2;
+  const unsigned int REQUESTLINE_HEADERS_MIN = 1;
```

Comment:
>I don't really see the use case

Logical organization which compliments design. In this case, the design is lacking (why you use arbitrary constants which are loosely related but not derived from a larger mathematical constant or why you choose to measure and implement data relationships haphazardly) so, I think learning organization is in your best interest.

These are some of the real issues to consider (not whether it's enum or not) but I need to digress until you're ready to listen and learn. @olarks did similar refactoring work (very well might I add), so he might also have some insight.

For now, let's just get this PR done with.

>I am just trying to avoid magic numbers.
?